tests/jerry/es.next/regression-test-issue-4150-main.mjs

```diff
@@ -0,0 +1,52 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import foo1 from "./regression-test-issue-4150-1.mjs"
+import foo2 from "./regression-test-issue-4150-2.mjs"
+import foo3 from "./regression-test-issue-4150-3.mjs"
+import foo4 from "./regression-test-issue-4150-4.mjs"
+
+let async_queue_expected = ["foo2", "foo3"];
+let async_queue = [];
+
+(function() {
+  assert(foo1().next().value === "foo1");
+})();
+
+(async function() {
+  async_queue.push((await foo2().next()).value);
+})();
+
+(async function() {
+  async_queue.push(await foo3());
+})();
+
+(function() {
+  assert(foo4 === "foo4");
+})();
+
+Array.prototype.assertArrayEqual = function(expected) {
+  assert(this.length === expected.length);
+
+  for (var i = 0; i < this.length; i++) {
+    assert(this[i] === expected[i]);
+  }
+}
+
+let global = new Function('return this;')();
+
+global.__checkAsync = function() {
+  async_queue.assertArrayEqual(async_queue_expected);
+}
+
```

Comment:
No need for the extra empty line at the end.

Reply:
Resolved!